src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,6 +166,38 @@ object DepositHandler {
     }
   }
 
+  private def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
+    log.debug(s"[$id] Checking fetch.txt")
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      { log.debug(s"[$id] Resolving files in fetch.txt")
+        fetchText.asScala.foreach(item =>
+        getUrl(item.getUrl).foreach(url =>
+          Using.urlInputStream(url).foreach(src =>
+            Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename)))))
+      }
+    )
+  }
+
```

Comment:
Handling `Options` and `Try's` with `map` and `foreach` is truly compact and neat. But I agree that it is not really intuitive when you have only one member or zero.